### DISCUSSION TITLE
View Templates

### DISCUSSION THREAD

**Discussion by Adriani6** [OP-TED/eForms-SDK#85] [Q&A] [answered]
Hi,

I'd like to apologise in advance if I'm not on the right path/this is not an issue and if I'm asking it in the wrong place!

I'm currently looking at view-template 29.efx and I'm using it with the example notice can_24_maximal.xml using the latest sdk version (1.0.0) and I'm having issues with a couple of lines in the view template.

I'd like to confirm whether the following lines (there might be more nodes, these are at the top of the file so easy to spot):

- 11: BT-105-Procedure
- 12: BT-106-Procedure
- 13: BT-1351-Procedure

Are in the wrong scope? Tracing back the relative paths does not yield the correct results, in fact the nodes for those fields do not appear in the context that the parent node is in, perhaps someone could shed some light?

Additionally, if view templates are not quite ready, is there one that's most up to date and working or are they all more or less in the same state?

Note, we are toying with the idea of rolling out our own implementation of the efx parser should that be of any help.

Regards,
Adrian

**Comment by rousso** [comment, marked as answer]
Hi @Adriani6,

On Friday 16/09 we released eforms SDK 1.1.0 which contains updated view templates. 

Many of the problems that you encountered may already have been resolved with this realease but the view templates are not finalized yet. 

The view templates are still a work in progress so we will keep working on them and keep releasing updates in the next several weeks. 

Creating your own EFX parser is not a bad idea. If the EFX toolkit for Java fits your needs you can alternatively use it to save some of the trouble of writing the entire thing yourselves. 

Are you working with Java? What template language/framework do you want to translate to? The notice viewer sample application for example translates to XSLT. Let me know and I may be able to help you decide which path to take. 


Regards,
Yannis

**Comment by Adriani6** [comment]
Hello @rousso,

Thanks for getting back to me. 
I'll give the latest release a shot, we've mainly had issues with connecting nodes using relative paths for some of the nodes.

We've considered using the EFX toolkit but we don't use Java and didn't want to introduce a dependency on the language just for this.

We're working with C# and we're looking at generating HTML mark up from the view-templates at the moment, however eventually (quite shortly), we'll also be working on creating notices using the SDK.

Thanks,
Adrian

**Comment by rousso** [comment]
Hi @Adriani6,

I will mark this question as answered because I think we covered your main inquiry. 
Please keep in touch about the work you are doing with EFX and give us feedback.

I am especially interested in the work you are doing in C# not only because many eSenders develop in C# but also because I used to be a C# developer myself for 20 years... 

We do not currently have the resources to do any meaningful work for porting the EFX Toolkit or the sample applications to C#. However I will support with every way I can the work that eSenders are doing in C# especially if some ports can be shared with other eSenders.

Regards,
Yannis
